saleor/order/migrations/tasks/saleor3_19.py

```diff
@@ -26,4 +31,117 @@ def update_order_addresses_task():
                 addresses.append(order_address)
         Address.objects.bulk_create(addresses, ignore_conflicts=True)
         Order.objects.bulk_update(orders, ["shipping_address"])
-        update_order_addresses_task.delay()
+        update_order_addresses_task()
+
+
+@app.task
+def set_udniscounted_base_shipping_price_on_orders_task():
+    qs = Order.objects.filter(undiscounted_base_shipping_price_amount__isnull=True)
+    order_ids = list(
+        qs.values_list("pk", flat=True)[:ORDER_SET_SHIPPING_PRICE_BATCH_SIZE]
+    )
+    if order_ids:
+        orders = Order.objects.filter(id__in=order_ids)
+
+        # get orders created from checkout that has shipping discount
+        # for draft orders the `base_shipping_price_amount` is the undiscounted price
+        # so we can use it as a base for the undiscounted price
+        orders_with_shipping_discount = _get_orders_with_shipping_discount(orders)
+
+        orders_no_shipping_discount = orders.exclude(
+            Exists(orders_with_shipping_discount.filter(pk=OuterRef("pk")))
+        )
+
+        if orders_no_shipping_discount:
+            _set_undiscounted_base_shipping_price(orders_no_shipping_discount)
+        if orders_with_shipping_discount:
+            _calculate_and_set_undiscounted_base_shipping_price(
+                orders_with_shipping_discount
+            )
+
+        set_udniscounted_base_shipping_price_on_orders_task.delay()
+
+
+def _get_orders_with_shipping_discount(orders):
+    orders_with_shipping_voucher = _get_orders_with_shipping_voucher(orders)
+    orders_with_shipping_voucher_no_voucher_instance = (
+        _get_orders_with_shipping_voucher_no_voucher_instance(orders)
+    )
+    return (
+        orders_with_shipping_voucher | orders_with_shipping_voucher_no_voucher_instance
+    )
+
+
+def _get_orders_with_shipping_voucher(orders):
+    shipping_vouchers = Voucher.objects.filter(type="shipping")
+    return orders.filter(
+        origin__in=["checkout"],
+        voucher_code__isnull=False,
+        voucher__isnull=False,
+    ).filter(Exists(shipping_vouchers.filter(pk=OuterRef("voucher_id"))))
+
+
+def _get_orders_with_shipping_voucher_no_voucher_instance(orders):
+    # lines with applied line voucher
+    lines_with_voucher = OrderLine.objects.filter(voucher_code__isnull=False)
+
+    # lines without applied order voucher on line
+    # this excludes the cases when entire order voucher was applied
+    # (for `ENTIRE_ORDER` voucher type, the voucher_code is stored on the order itself,
+    # not on the line, but the discount is propagated to the line level and it's visible
+    # in the `unit_price`)
+    # - `base_unit_price_amount` is the price with the sale and line voucher applied
+    # - `unit_price` is the price with all discounts applied - sales,
+    # line and order voucher
```

Comment:
Did you take into account manual discounts, which decrease a value of unit price?

Reply:
Manual discount are only applicable on orders created from draft orders, and for such orders we can copy value from `base_shipping_price` as this is actually `undiscounted_base_shipping_price`